src/framegrab/rtsp_discovery.py

```diff
@@ -0,0 +1,150 @@
+        """
+        Uses WSDiscovery to find ONVIF supported devices.
+
+        Parameters:
+        try_default_logins (bool, optional): Option to try different default credentials stored in DEFAULT_CREDENTIALS.
+            Defaults to False.
+
+        Returns:
+        List[ONVIFDeviceInfo]: A list of ONVIFDeviceInfos with IP address, port number, and ONVIF service address.
+        """
+
+        device_ips = []
+        logger.debug("Starting WSDiscovery for ONVIF devices")
+        wsd = WSDiscovery()
+        wsd.start()
+        types = [QName("http://www.onvif.org/ver10/network/wsdl", "NetworkVideoTransmitter")]
+        ret = wsd.searchServices(types=types)
+        for service in ret:
+            xaddr = service.getXAddrs()[0]
+            parsed_url = urllib.parse.urlparse(xaddr)
+            ip = parsed_url.hostname
+            port = parsed_url.port or 80  # Use the default port 80 if not specified
+
+            logger.debug(f"Found ONVIF service at {xaddr}")
+            device_ip = ONVIFDeviceInfo(ip=ip, port=port, username="", password="", xaddr=xaddr, rtsp_urls=[])
+
+            if try_default_logins:
+                RTSPDiscovery._try_logins(device=device_ip)
+
+            device_ips.append(device_ip)
+        wsd.stop()
+        return list(device_ips)
+
+    @staticmethod
+    def generate_rtsp_urls(device: ONVIFDeviceInfo) -> bool:
+        """
+        Fetch RTSP URLs from an ONVIF supported device, given a username/password.
+
+        Parameters:
+        device (ONVIFDeviceInfo): Pydantic Model that stores information about camera RTSP address, port number, username, and password.
+
+        Returns:
+        bool: False if the device is unreachable or the credentials are wrong, else returns True and updates ONVIFDeviceInfo with updated rtsp_urls.
+        """
+
+        rtsp_urls = []
+        try:
+            try:
+                # Assuming port 80, adjust if necessary
+                cam = ONVIFCamera(device.ip, device.port, device.username, device.password)
+                # Create media service
+                media_service = cam.create_media_service()
+                # Get profiles
+                profiles = media_service.GetProfiles()
+                stream_setup = {
+                    "Stream": "RTP-Unicast",  # Specify the type of stream
+                    "Transport": {"Protocol": "RTSP"},
+                }
+
+                # For each profile, get the RTSP URL
+                for profile in profiles:
+                    stream_uri = media_service.GetStreamUri(
+                        {"StreamSetup": stream_setup, "ProfileToken": profile.token}
+                    )
+                    rtsp_urls.append(stream_uri.Uri)
+            except onvif.exceptions.ONVIFError as e:
+                msg = str(e).lower()
+                if "auth" in msg:  # looks like a bad login - give up.
+                    return False
+                else:
+                    raise  # something else
```

Comment:
should probably raise a specific exception. Maybe the original exception of `onvif.exepctions.ONVIFError`